src/nanoemoji/png.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from pathlib import Path
+from typing import Union
+
+
+class PNG(bytes):
+
+    # The first eight bytes of a PNG file always contain the following (decimal) values:
+    #   137 80 78 71 13 10 26 10
+    # https://www.w3.org/TR/PNG-Structure.html
+    SIGNATURE = b"\x89PNG\r\n\x1a\n"
+
+    def __new__(cls, *args, **kwargs):
+        self = super().__new__(cls, *args, **kwargs)
+        header = self[:8]
+        if header != cls.SIGNATURE:
+            raise ValueError("Invalid PNG file: bad signature {header!r}")
+        return self
+
+    @classmethod
+    def read_from(cls, path: Union[str, os.PathLike]) -> "PNG":
+        return cls(Path(path).read_bytes())
```

Comment:
nit: reading the whole file to get 8 bytes seems a touch generous with memory even for Python

Reply:
I'm not reading it just to get 8 bytes, we are reading all bytes anyway in order to stick them in the CBDT/sbix tables, we may as well also assert that it's actually a valid PNG file.

Reply:
Maybe you meant that we should not reading all PNGs upfront upon gathering the input glyphs, but only one by one as we build the bitmap tables glyphs, releasing them afterwards to avoid keeping them all in memory throughout the build? I did that to keep consistent with the pattern where InputGlyph as well as ColorGlyph hold both an svg filename and a preloaded SVG object, not just the filename. But we aren't doing much to the bitmaps, it's true, we can change it if your prefer.

Reply:
Oh my bad, I read it as doing nothing other than reading the identifying bytes and missed we are hanging onto them to shove into bitmap tables. False alarm!